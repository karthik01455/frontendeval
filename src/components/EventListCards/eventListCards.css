.event-list-card {
  display: flex;
  height: 400px;
  width: 300px;
  flex-direction: column;
  border-radius: 3%;
  overflow: hidden;
  background-color: black;
  color: white;
}
.event-list-image-container {
  height: 200px;
  width: 300px;
}
.event-list-image {
  height: 200px;
  width: 300px;
  border: 0 0 5px 0 solid white;
}

.event-list-card-text {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.event-list-card-name {
  display: flex;
  margin: 3% 5% 3% 5%;
  font-weight: bold;
}
.event-list-card-description {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  justify-content: center;
  -webkit-box-orient: vertical;
  overflow: hidden;
  margin: 3% 8% 3% 8%;
}
.event-list-card-text-footer {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  font-size: small;
}
.event-list-card-footer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 3% 8% 3% 8%;
}
.event-list-card-seat {
}
.event-list-card-registered {
  display: flex;
  row-gap: 2%;
  flex-direction: row;
}
.event-list-card-bookmark {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  overflow: hidden;
}
