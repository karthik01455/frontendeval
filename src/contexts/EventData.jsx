/* eslint-disable react/prop-types */
import React, { createContext, useState } from 'react';

export const EventDataContext = createContext({});

export function EventDataProvider({ children }) {
  const [eventData, setEventData] = useState(null);
  const [search, setSearch] = useState(null);

  return (
    <EventDataContext.Provider
      value={{
        eventData,
        setEventData,
        search,
        setSearch,
      }}
    >
      {children}
    </EventDataContext.Provider>
  );
}
