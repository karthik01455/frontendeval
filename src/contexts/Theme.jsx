/* eslint-disable react/prop-types */
import React, { createContext, useState } from 'react';

export const ThemeContext = createContext({});

export function ThemeProvider({ children }) {
  const [theme, setTheme] = useState(null);

  return (
    <ThemeContext.Provider
      value={{
        theme,
        setTheme,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
}
