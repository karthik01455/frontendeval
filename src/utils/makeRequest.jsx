import axios from 'axios';
import { BACKEND_URL } from '../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig = {}, navigate) => {
  const requestDetails = {
    baseURL: BACKEND_URL,
    url: apiEndPoint.url,
    method: apiEndPoint.method,
    ...dynamicConfig,
    // headers: {
    //   authorization: "Bearer QWlzaHdhcnlhIE4=",
    // },
  };
  try {
    console.log(`req${JSON.stringify(requestDetails)}`);
    const { data } = await axios(requestDetails);
    console.log('hi');
    console.log(`data-mr${JSON.stringify(data)}`);
    return data;
  } catch (e) {
    if (navigate) {
      console.log('es1', e.response);
      const errorStatus = e.response?.status;
      console.log('es', errorStatus);
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};
export default makeRequest;
