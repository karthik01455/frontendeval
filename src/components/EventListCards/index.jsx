/* eslint-disable react/prop-types */
import React, { useContext, useState } from 'react';
import makeRequest from '../../utils/makeRequest';
import { EventDataContext } from '../../contexts/EventData';
import { ThemeContext } from '../../contexts/Theme';
('../../utils/makeRequest');
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useNavigate } from 'react-router-dom';
import './eventListCards.css';
import { PATCH_EVENTS } from '../../constants/apiEndPoints';
import {
  faBookmark,
  faCircleXmark,
  faBookBookmark,
  faCircleCheck,
} from '@fortawesome/free-solid-svg-icons';

export default function EventListCards({ value }) {
  const navigate = useNavigate();
  const date = new Date(value.datetime);
  const months = [
    'Jan',
    'Feb',
    'Mar',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  let month = months[date.getMonth()];
  const dateEvent = date.getDate();
  const year = date.getFullYear();
  const timeZone = date.getTimezoneOffset();
  const time = date.getTime();
  const [bookMarked, setBookMarked] = useState(value.isBookmarked);
  const { eventData, setEventData } = useContext(EventDataContext);
  const { event, setEvent } = useContext(EventDataContext);
  const { theme, setTheme } = useContext(ThemeContext);
  return (
    <div
      data-testid='card'
      className='el-card'
      key={value.id}
      style={{ backgroundColor: theme }}
    >
      <div
        className='el-image-container'
        onClick={() => {
          setEvent(value);
          navigate(`/eventDetails`);
        }}
      >
        <img className='el-image' src={value.imgUrl}></img>
      </div>

      <div
        className='el-card-text'
        onClick={() => {
          setEvent(value);
          navigate(`/eventDetails`);
        }}
      >
        <div className='el-card-name'>{value.name}</div>
        <div className='el-card-description'>{value.description}</div>
        <div className='el-card-text-footer'>
          <div>VENUE:{value.venue}</div>
          <div>
            DATE:{dateEvent} {month} {year} {time}
          </div>
        </div>
      </div>
      <div className='el-card-footer'>
        <div className='el-card-registered'>
          {/* <FontAwesomeIcon icon={value.isRegistered ? faCircleCheck : null} /> */}
          {/* <div>{value.isRegistered ? 'REGISTERED' : null}</div> */}
          <div className='el-card-seat'>
            <FontAwesomeIcon
              icon={
                !value.areSeatsAvailable && !value.isRegistered
                  ? faCircleXmark
                  : faCircleCheck
              }
            />
          </div>

          <div>
            {!value.isRegistered && !value.areSeatsAvailable
              ? 'NO SEATS AVAILABLE'
              : null}
          </div>
        </div>
        <div
          className='el-card-bookmark'
          onClick={() => {
            makeRequest(PATCH_EVENTS(value.id), {
              data: { isBookmarked: !bookMarked },
            }).then((res) => {
              const eventDataChange = [...eventData];
              const recordIndex = eventDataChange.findIndex(
                (record) => record.id === res.id
              );
              eventDataChange[recordIndex]
                ? (eventDataChange[recordIndex].isBookmarked =
                    !eventDataChange[recordIndex].isBookmarked)
                : null;
              setEventData(eventDataChange);
              setBookMarked(!bookMarked);
            });
          }}
        >
          <FontAwesomeIcon icon={bookMarked ? faBookmark : faBookBookmark} />
        </div>
      </div>
    </div>
  );
}
