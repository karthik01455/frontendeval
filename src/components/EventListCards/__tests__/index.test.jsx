import React from 'react';
import { fireEvent, screen, render, waitFor } from '@testing-library/react';
import EventListCards from '..';
import makeRequest from '../../../utils/makeRequest';
import { mockData } from '../../../mocks/mockData';
import { ThemeContext } from '../../../contexts/Theme';
import { EventDataContext } from '../../../contexts/EventData';
const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockedNavigate,
}));
jest.mock('../../../utils/makeRequest');
describe('EvenListCards', () => {
  test('should render the same snapshot when it is called ', () => {
    makeRequest.mockResolvedValue(mockData);
    const { container } = render(
      <EventDataContext.Provider
        value={{
          bookData: null,
          setBookData: jest.fn(),
        }}
      >
        <EventListCards value={mockData[0]} />
      </EventDataContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
describe('EvenListCards', () => {
  test('should render the card correctly when correct data is passed ', () => {
    makeRequest.mockResolvedValue(mockData);
    const { container } = render(
      <EventDataContext.Provider
        value={{
          bookData: null,
          setBookData: jest.fn(),
        }}
      >
        <EventListCards value={mockData[0]} />
      </EventDataContext.Provider>
    );
    expect(screen.getByTestId('card')).toBeTruthy();
  });
});
