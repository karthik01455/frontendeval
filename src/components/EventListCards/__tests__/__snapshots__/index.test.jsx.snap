// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvenListCards should render the same snapshot when it is called  1`] = `
<div>
  <div
    class="el-card"
    data-testid="card"
  >
    <div
      class="el-image-container"
    >
      <img
        class="el-image"
        src="https://i.ibb.co/3zbdvWX/battle-of-bands.jpg"
      />
    </div>
    <div
      class="el-card-text"
    >
      <div
        class="el-card-name"
      >
        Battle of the Bands
      </div>
      <div
        class="el-card-description"
      >
        Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won't want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don't miss out on the most unforgettable rock competition of the year!
      </div>
      <div
        class="el-card-text-footer"
      >
        <div>
          VENUE:
          All Stars Arena, Las Vegas, NV, USA
        </div>
        <div>
          DATE:
          1
           
          Mar
           
          2023
           
          1677646800000
        </div>
      </div>
    </div>
    <div
      class="el-card-footer"
    >
      <div
        class="el-card-registered"
      >
        <div
          class="el-card-seat"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-check "
            data-icon="circle-check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div />
      </div>
      <div
        class="el-card-bookmark"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-book-bookmark "
          data-icon="book-bookmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 43 43 0 96 0h96V190.7c0 13.4 15.5 20.9 26 12.5L272 160l54 43.2c10.5 8.4 26 .9 26-12.5V0h32 32c17.7 0 32 14.3 32 32V352c0 17.7-14.3 32-32 32v64c17.7 0 32 14.3 32 32s-14.3 32-32 32H384 96c-53 0-96-43-96-96V96zM64 416c0 17.7 14.3 32 32 32H352V384H96c-17.7 0-32 14.3-32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
